myhpom/tests/test_signup_view.py

```diff
@@ -0,0 +1,60 @@
+from django.core.urlresolvers import reverse
+from django.test import Client, TestCase
+from myhpom.models import User, UserDetails, State
+
+
+class SignupTestCase(TestCase):
+    """
+    * GET retries page
+    * POST invalid data returns to the page
+    * POST valid data in non-supported state for non-existing user redirects to 'next_steps'
+    * POST valid data in supported state for non-existing user redirects to 'choose_network'
+    """
+
+    def setUp(self):
+        self.client = Client()
+        self.url = reverse('myhpom:signup')
+        self.form_data = {
+            'first_name': 'A',
+            'last_name': 'B',
+            'email': 'ab@example.com',
+            'password': 'Abbbbbbb1@',
+            'password_confirm': 'Abbbbbbb1@',
+            'state': 'NC',  # a supported state
+            'accept_tos': True,
+        }
+
+    def test_get_signup(self):
+        """GET the signup page should work"""
+        response = self.client.get(self.url)
+        self.assertEqual(200, response.status_code)
+        self.assertTemplateUsed('myhpom/accounts/signup.html')
+
+    def test_post_signup_invalid(self):
+        """invalid signup should redisplay signup page"""
+        data = {}  # empty signup data is certainly invalid!
+        response = self.client.post(self.url, data=data)
+        self.assertEqual(200, response.status_code)
+        self.assertTemplateUsed('myhpom/accounts/signup.html')
```

Comment:
assertions on `form.errors` in the `response.context` would make sense here - we would expect no data to result in some errors!